### ISSUE TITLE
Pinning and sandboxed network access

### ISSUE THREAD

**Issue by dbuenzli** [ocaml/opam#3771]
Some packages like [`uucp`](https://github.com/dbuenzli/uucp) when pinned need network access to download additional data files that are not needed in distribution builds.

The new sandbox feature prevents this package from being pinned. I'm actually very happy the sanboxing works but in this case it's a bit annoying me.

It seems the  [`extra-source:`](https://opam.ocaml.org/doc/Manual.html#opamsection-extra-sources) field could be used but it doesn't allow to specify a filter e.g. on `dev` or `pinned`. The static nature of it may also be a bit limiting. 

Basically we lost a simple workflow here where I could instruct people to `opam pin --dev` to test updates to the package with no additional friction.

Would it be possible to selectively disable the network sandboxing on an single `pin add/ install/resintall/upgrade`. Something like an `--allow-network-access` option that would affect only the exact packages mentioned on the cli (not the deps) ?

**Comment by dbuenzli** [comment]
Aside -- I'm actually tempted to change the my practice here and simply integrate the data file in the repo itself, maybe it makes things simpler, just that I'm not too happy to add a 45Mo xml file to that repo every year.

**Comment by avsm** [comment]
One option here is to add the large data files to a git branch, so they aren't fetched into the `master` branch. You could set `dev-repo` accordingly in the opam metadata.

I'm very much not in favour of disabling networking sandboxing if at all possible...

**Comment by dbuenzli** [comment]
It turns out that it's not the only package that would need this 45Mo annual data file (the way things are structured now `uunf` needs it aswell). 

Yet another option is to make a specific repo with these data files and make that repo an out of ocaml opam-repository `dev` dependency. But on the other hand I'm not too keen on mirroring the ucd and test data files from https://www.unicode.org/Public/UCD in a git repo.

**Comment by avsm** [comment]
Why does the dev dependency need to be a git repo? It could just be a versioned tarball package that is only pulled down if the OCaml package is pinned

**Comment by dbuenzli** [comment]
> Why does the dev dependency need to be a git repo? It could just be a versioned tarball package that is only pulled down if the OCaml package is pinned

Well I'd still need to create the whole thing myself and upload it somewhere; at that point a `git` repo is easier... Having `dev` filters on `extra-source:` would allow to directly pull from the unicode website on pins though formally you may want to select the version dynamically after a configure step.

**Comment by dbuenzli** [comment]
This is a niche usage, workarounds can be found. Let's close this.
